### PR TITLE
Modify rule S2701: promote C# to SonarWay

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
S2701 CSharp:
- RSPEC cleanup
- Promote rule to `SonarWay` profile

### CODE REVIEW

File: rules/S2701/csharp/rule.adoc [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
-(visible only on this page)
+include::how-nunit.adoc[]
 
-include::../message.adoc[]
+include::how-xunit.adoc[]
 
-'''
-== Comments And Links
-(visible only on this page)
+include::../rspecator.adoc[]
```
Comment (sebastien-marichal) [inline rules/S2701/csharp/rule.adoc:17]: Probably not a big issue, I think this should be after the Resources section

File: rules/S2701/csharp/rule.adoc [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
 
-[source,csharp]
-----
-bool b = true;
-NUnit.Framework.Assert.AreEqual(true, b);
-Xunit.Assert.NotSame(true, b);
-Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual(true, b);
-System.Diagnostics.Debug.Assert(true);
-----
+`Assert.True` is not raising this issue for xUnit because `Assert.Fail` was introduced in 2020 version 2.4.2 and `Assert.True(false, message)` was a way to simulate it.
```
Comment (sebastien-marichal) [inline rules/S2701/csharp/rule.adoc:9]: Just to be sure, do we only ignore `Assert.True(false)` and not `Assert.False(true)` as well?

**Review by sebastien-marichal — APPROVED** [review]
LGTM!
Left only nitpick comments

File: rules/S2701/csharp/rule.adoc (hunk 2 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S2701/csharp/rule.adoc:9]: Exactly, `AssertFalse(true, ...)` will report, this is at least the current rule implementation

File: rules/S2701/csharp/rule.adoc (hunk 1 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S2701/csharp/rule.adoc:17]: Indeed!

File: rules/S2701/csharp/rule.adoc (hunk 2 above)
Comment (sebastien-marichal) [reply rules/S2701/csharp/rule.adoc:9]: Having both ignored could save a few FPs maybe, WDYT?
Look like to me a low hanging FP.

Comment (cristian-ambrosini-sonarsource) [reply rules/S2701/csharp/rule.adoc:9]: On Sourcegraph `Assert.True(false` reports 7.2k+ results, while `Assert.False(true` returns 2.9k. I think during the implementation [this comment](https://github.com/xunit/xunit/issues/2027#issuecomment-549434940) was the prominent source for this and we skipped only Assert.True. Maybe you are right and we should play safe and avoid reporting on Assert.False. We will create some FN as a drawback. I will update this rspec and do the change in sonar-dotnet

Comment (sebastien-marichal) [reply rules/S2701/csharp/rule.adoc:9]: In which case `Assert.False(true)` would be an FN?

Comment (cristian-ambrosini-sonarsource) [reply rules/S2701/csharp/rule.adoc:9]: Assert.False(false,..)

Comment (cristian-ambrosini-sonarsource) [reply rules/S2701/csharp/rule.adoc:9]: https://github.com/SonarSource/sonar-dotnet/pull/8823

File: rules/S2701/csharp/rule.adoc [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
-bool b = true;
-NUnit.Framework.Assert.AreEqual(true, b);
-Xunit.Assert.NotSame(true, b);
-Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual(true, b);
-System.Diagnostics.Debug.Assert(true);
-----
+`Assert.True` and `Assert.False` are not raising this issue for xUnit because `Assert.Fail` was introduced in 2020 version 2.4.2 and both `Assert.True(false, message)` and `Assert.False(true, message)` were ways to simulate it.
 
-ifdef::env-github,rspecator-view[]
+=== Exceptions
```
Comment (sebastien-marichal) [inline rules/S2701/csharp/rule.adoc:11]: There is 2 Exceptions sections

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]

File: rules/S2701/csharp/rule.adoc (hunk 3 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S2701/csharp/rule.adoc:11]: There you go, thanks for taking a second look 😄

**Review by sebastien-marichal — APPROVED** [review]
LGTM!

**Pull request merged**